Felienne/hedy
# Add runtime validation and correction to the turtle commands transpiled code #1983

**Please fill out this template by replacing all content between <>**

Add runtime validation and correction to the turtle commands transpiled code

**Description**

- When the turn or forward commands are passed large positive or negative numbers as arguments, their input is capped to a number that does not cause failure. The check is done at runtime which means that the cap should work even when the argument is passed via user input.
- When the turtle commands receive a value that cannot be converted to a number, an error message is displayed. This error message cannot be translated and it is only in English.

**Fixes #1983 **

< Always link the number of the issue or of the discussion that your pr concerns. >
Notable exception: adding content may be done without an accompanying issue 

**How to test**

Run Hedy locally and execute the following programs:
- In level 1 run ```forward 99999999```, ```forward -99999999999```, ```turn 99999999999``` and ```turn -9999999999```. Since the large numbers will be capped to 600 and -600, respectively, the turtle will probably go out of your screen. However, your browser should not hang and the program should only take a couple of secs before it completes successfully.

- In level 4 run:
```
input is 999999999, -999999999
turn input at random
forward input at random
```
Again, the turtle will probably get out of your screen but the program should not hang and it should complete successfully.

- In level 4 execute:
```
input is not, a, number
turn input at random
forward input at random
```
Ensure that you get an error. Is the error clear? Does it give enough directions to fix the program?

**Checklist**
Done? Check if you have it all in place using this list*
  

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
